### ISSUE TITLE
Improved GUI for SaltStack

### ISSUE THREAD

**Issue by DemiMarie** [QubesOS/qubes-issues#4662]
### Qubes OS version:

`Qubes release 4.0 (R4.0)`

### Affected component(s): 

dom0 updates GUI

---

### Steps to reproduce the behavior:

Run dom0 updates GUI

### Expected behavior:

The output of each VM is displayed in a separate text box, with highlighting.  Furthermore, arbitrary Salt parameters can be passed.  Finally, copying of output that is not from dom0 is blocked.

### Actual behavior:

The outputs of each VM are displayed in a shared text box, without coloring.  Only the Salt update command can be run.

### General notes:

We only need to worry about the subset of ANSI control codes produced by SaltStack, which is essentially trivial.

---

### Related issues:

#4543, but this issue is far simpler regarding its security considerations.

**Comment by marmarek** [comment]
> Only the Salt update command can be run.

This is on purpose. This isn't generic GUI for SaltStack, it's a GUI for updates. So, this ticket should rather be named "Feature request: add generic GUI for SaltStack".

**Comment by ben-grande** [comment]
@DemiMarie Do you think it's still relevant seen that there is a new updater interacting with package managers in qubes without Salt?

**Comment by DemiMarie** [comment]
@ben-grande I don’t think this is needed.  Also, I expect that Salt is on its way out in Qubes OS due to not supporting distro packaging very well and having a lot of complexity.

**Comment by andrewdavidwong** [comment]
> [@ben-grande](https://github.com/ben-grande) I don’t think this is needed. Also, I expect that Salt is on its way out in Qubes OS due to not supporting distro packaging very well and having a lot of complexity.

Meaning this issue should be closed?

**Comment by DemiMarie** [comment]
> > [@ben-grande](https://github.com/ben-grande) I don’t think this is needed. Also, I expect that Salt is on its way out in Qubes OS due to not supporting distro packaging very well and having a lot of complexity.
> 
> Meaning this issue should be closed?

Correct.
